### PR TITLE
Servicebus subscription tf

### PR DESCRIPTION
**Pull request by vijayrajagopalan-hmcts**
### JIRA link (if applicable) ###
Upgrade service bus subscription module

### Change description ###
Upgrade service bus subscription module

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,38 @@
...
+
+[Azure Service Bus Subscription](https://docs.microsoft.com/en-us/azure/service-bus-messaging/service-bus-queues-topics-subscriptions) <br />
+
+## Usage
+
+The following example shows how to use the module to create an Azure Service Bus subscription. 
+
+```terraform
+module "servicebus-subscription" {
+  source                = "git@github.com:hmcts/terraform-module-servicebus-subscription?ref=servicebus_subscription_tf"
```
Comment (timja) [inline README.md:14]:
```suggestion
  source                = "git@github.com:hmcts/terraform-module-servicebus-subscription?ref=master"
```

File: README.md [hunk 2]
```diff
@@ -1,3 +1,38 @@
...
+  name                  = local.subscription_name
```
Comment (timja) [inline README.md:15]:
```suggestion
  name                  = "your-subscription"
```

File: README.md [hunk 3]
```diff
@@ -1,3 +1,38 @@
...
+  namespace_name        = module.servicebus-namespace.name
+  topic_name            = module.servicebus-topic.name
+  resource_group_name   = local.resource_group_name
```
Comment (timja) [inline README.md:18]:
```suggestion
  resource_group_name   = azurerm_resource_group.rg.name
```

File: main.tf [hunk 4]
```diff
@@ -1,22 +1,17 @@
...
-  parameters = {
-    serviceBusNamespaceName       = "${var.namespace_name}"
-    serviceBusTopicName           = "${var.topic_name}"
-    serviceBusSubscriptionName    = "${var.name}"
-    lockDuration                  = "${var.lock_duration}"
-    maxDeliveryCount              = "${var.max_delivery_count}"
-    forwardTo                     = "${var.forward_to}"
-    forwardDeadLetteredMessagesTo = "${var.forward_dead_lettered_messages_to}"
-  }
+  requires_session                     = false
```
Comment (timja) [inline main.tf:12]: you created a variable for this but aren't using it
```suggestion
  requires_session                     = var.requires_session
```

File: main.tf [hunk 5]
```diff
@@ -1,22 +1,17 @@
...
+  dead_lettering_on_message_expiration = true
```
Comment (timja) [inline main.tf:13]: do either this var or the next one need to be configurable?

**Review by timja — COMMENTED** [review]
few minor comments

File: main.tf (hunk 5 above)
Comment (vijayrajagopalan-hmcts) [reply main.tf:13]: > do either this var or the next one need to be configurable?

@timja requires_session is now configurable. dead_lettering_on_message_expiration in the existing configuration is hardcoded as true. Happy to make this configurable. Could you pls confirm?

Comment (timja) [reply main.tf:13]: probably is fine to leave it as is

**Comment by timja** [comment]
you have conflicts which need resolving with master

File: main.tf (hunk 5 above)
Comment (vijayrajagopalan-hmcts) [reply main.tf:13]: Ok

**Review by timja — APPROVED** [review]

**Comment by vijayrajagopalan-hmcts** [comment]
This PR will be merged after the issues in AAT are resolved to ensure the import and build executes successfully.

@timja @adusumillipraveen

**Pull request merged**